src/Core/Framework/App/Subscriber/SystemLanguageChangedSubscriber.php

```diff
@@ -0,0 +1,113 @@
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Maintenance\System\Service\SystemLanguageChangeEvent;
+use Shopware\Core\System\Language\LanguageCollection;
+use Shopware\Core\System\Language\LanguageEntity;
+use Shopware\Core\System\Locale\LocaleEntity;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+readonly class SystemLanguageChangedSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @param EntityRepository<LanguageCollection> $languageRepository
+     * @param EntityRepository<AppCollection> $appRepository
+     * @param EntityRepository<AppAdministrationSnippetCollection>|null $snippetRepository
+     */
+    public function __construct(
+        private EntityRepository $languageRepository,
+        private EntityRepository $appRepository,
+        private ?EntityRepository $snippetRepository,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            SystemLanguageChangeEvent::class => 'onSystemLanguageChanged',
+        ];
+    }
+
+    public function onSystemLanguageChanged(SystemLanguageChangeEvent $event): void
+    {
+        // This repository is part of the Administration bundle and not necessarily installed and available.
+        if ($this->snippetRepository === null) {
+            return;
+        }
+
+        $context = Context::createDefaultContext();
+
+        $snippets = $this->getSnippets($this->snippetRepository, $context);
+        if ($snippets->count() === 0) {
+            return;
+        }
+
+        $appIds = $this->getApps($context)->getIds();
+
+        $appsWithSnippets = [];
+        foreach ($snippets as $snippet) {
+            if (\in_array($snippet->getAppId(), $appIds, true) && !\in_array($snippet->getAppId(), $appsWithSnippets, true)) {
+                $appsWithSnippets[] = $snippet->getAppId();
+            }
+        }
+
+        $previousLocale = $this->getLocale($event->previousLanguageId, $context);
+        $newLocale = $this->getLocale(Defaults::LANGUAGE_SYSTEM, $context);
+
+        foreach ($appsWithSnippets as $appId) {
+            $snippetToClone = $snippets->filter(fn (AppAdministrationSnippetEntity $snippet) => $appId === $snippet->getAppId() && $snippet->getLocaleId() === $newLocale->getId())->first();
+            \assert($snippetToClone instanceof AppAdministrationSnippetEntity);
+
+            $snippetWithPreviousLocaleExists = $snippets->filter(fn (AppAdministrationSnippetEntity $snippet) => $appId === $snippet->getAppId() && $snippet->getLocaleId() === $previousLocale->getId())->first();
+            if ($snippetWithPreviousLocaleExists) {
+                continue;
+            }
+
+            $this->snippetRepository->create([[
```

Comment:
Need to upsert here

Reply:
Hmm not sure, I think in no case we want to override existing snippets - only if they do not exist, we need to create them 🤔